### ISSUE TITLE
Pre-trained models and the number of epochs to reproduce results.

### ISSUE THREAD

**Issue by Yang-Li-2000** [Jun-CEN/Open_world_3D_semantic_segmentation#16]
Hi,

Could you please provide pre-trained models trained using ood_basic, ood_final, and ood_incre on the semantic kitti dataset?

We are trying to reproduce the results, but the mIoU we obtained after training is lower than expected. Specifically, at the beginning of ood_final, the valid mIoU measured by the code is 58, which is consistent with the number in the paper. However, after training using ood_final for one epoch, the mIoU dropped to 37.5. At the end of epoch 5, the mIoU is 45. 

How many epochs should I train to reproduce the results in the paper (64.3) for the semantic kitti dataset? 

Meanwhile, is it possible to release the checkpoints?

**Comment by luoyuchenmlcv** [comment]
Could you please also release the nusc checkpoints? Thanks!

**Comment by Jun-CEN** [comment]
Hi,

Checkpoints are available [here](https://github.com/Jun-CEN/Open_world_3D_semantic_segmentation/blob/084c341291ae186f174743d8fae5efecea89122f/README.md?plain=1#L62).

**Comment by Yang-Li-2000** [comment]
Hi Dr. Cen,

I tried to load semantic kitti checkpoints using val_cylinder_asym_ood.py. However, it seems that there are many size mismatches. 



Loading ../../semantic_kitti/checkpoints/downloaded/semantickitti_ood_final.pt
No match: cylinder_3d_spconv_seg.downCntx.conv1.weight Shape in my_modeltorch.Size([32, 1, 3, 3, 16]).    Shape in checkpoint: torch.Size([1, 3, 3, 16, 32])
No match: cylinder_3d_spconv_seg.downCntx.conv1_2.weight Shape in my_modeltorch.Size([32, 1, 3, 3, 32]).    Shape in checkpoint: torch.Size([3, 1, 3, 32, 32])
No match: cylinder_3d_spconv_seg.downCntx.conv2.weight Shape in my_modeltorch.Size([32, 3, 1, 3, 16]).    Shape in checkpoint: torch.Size([3, 1, 3, 16, 32])
No match: cylinder_3d_spconv_seg.downCntx.conv3.weight Shape in my_modeltorch.Size([32, 3, 1, 3, 32]).    Shape in checkpoint: torch.Size([1, 3, 3, 32, 32])
No match: cylinder_3d_spconv_seg.resBlock2.conv1.weight Shape in my_modeltorch.Size([64, 3, 1, 3, 32]).    Shape in checkpoint: torch.Size([3, 1, 3, 32, 64])
No match: cylinder_3d_spconv_seg.resBlock2.conv1_2.weight Shape in my_modeltorch.Size([64, 3, 1, 3, 64]).    Shape in checkpoint: torch.Size([1, 3, 3, 64, 64])
No match: cylinder_3d_spconv_seg.resBlock2.conv2.weight Shape in my_modeltorch.Size([64, 1, 3, 3, 32]).    Shape in checkpoint: torch.Size([1, 3, 3, 32, 64])
No match: cylinder_3d_spconv_seg.resBlock2.conv3.weight Shape in my_modeltorch.Size([64, 1, 3, 3, 64]).    Shape in checkpoint: torch.Size([3, 1, 3, 64, 64])
No match: cylinder_3d_spconv_seg.resBlock2.pool.weight Shape in my_modeltorch.Size([64, 3, 3, 3, 64]).    Shape in checkpoint: torch.Size([3, 3, 3, 64, 64])
No match: cylinder_3d_spconv_seg.resBlock3.conv1.weight Shape in my_modeltorch.Size([128, 3, 1, 3, 64]).    Shape in checkpoint: torch.Size([3, 1, 3, 64, 128])
No match: cylinder_3d_spconv_seg.resBlock3.conv1_2.weight Shape in my_modeltorch.Size([128, 3, 1, 3, 128]).    Shape in checkpoint: torch.Size([1, 3, 3, 128, 128])
No match: cylinder_3d_spconv_seg.resBlock3.conv2.weight Shape in my_modeltorch.Size([128, 1, 3, 3, 64]).    Shape in checkpoint: torch.Size([1, 3, 3, 64, 128])
No match: cylinder_3d_spconv_seg.resBlock3.conv3.weight Shape in my_modeltorch.Size([128, 1, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 1, 3, 128, 128])
No match: cylinder_3d_spconv_seg.resBlock3.pool.weight Shape in my_modeltorch.Size([128, 3, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 128])
No match: cylinder_3d_spconv_seg.resBlock4.conv1.weight Shape in my_modeltorch.Size([256, 3, 1, 3, 128]).    Shape in checkpoint: torch.Size([3, 1, 3, 128, 256])
No match: cylinder_3d_spconv_seg.resBlock4.conv1_2.weight Shape in my_modeltorch.Size([256, 3, 1, 3, 256]).    Shape in checkpoint: torch.Size([1, 3, 3, 256, 256])
No match: cylinder_3d_spconv_seg.resBlock4.conv2.weight Shape in my_modeltorch.Size([256, 1, 3, 3, 128]).    Shape in checkpoint: torch.Size([1, 3, 3, 128, 256])
No match: cylinder_3d_spconv_seg.resBlock4.conv3.weight Shape in my_modeltorch.Size([256, 1, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 1, 3, 256, 256])
No match: cylinder_3d_spconv_seg.resBlock4.pool.weight Shape in my_modeltorch.Size([256, 3, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 3, 3, 256, 256])
No match: cylinder_3d_spconv_seg.resBlock5.conv1.weight Shape in my_modeltorch.Size([512, 3, 1, 3, 256]).    Shape in checkpoint: torch.Size([3, 1, 3, 256, 512])
No match: cylinder_3d_spconv_seg.resBlock5.conv1_2.weight Shape in my_modeltorch.Size([512, 3, 1, 3, 512]).    Shape in checkpoint: torch.Size([1, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.resBlock5.conv2.weight Shape in my_modeltorch.Size([512, 1, 3, 3, 256]).    Shape in checkpoint: torch.Size([1, 3, 3, 256, 512])
No match: cylinder_3d_spconv_seg.resBlock5.conv3.weight Shape in my_modeltorch.Size([512, 1, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 1, 3, 512, 512])
No match: cylinder_3d_spconv_seg.resBlock5.pool.weight Shape in my_modeltorch.Size([512, 3, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock0.trans_dilao.weight Shape in my_modeltorch.Size([512, 3, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock0.conv1.weight Shape in my_modeltorch.Size([512, 1, 3, 3, 512]).    Shape in checkpoint: torch.Size([1, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock0.conv2.weight Shape in my_modeltorch.Size([512, 1, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 1, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock0.conv3.weight Shape in my_modeltorch.Size([512, 1, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock0.up_subm.weight Shape in my_modeltorch.Size([512, 3, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 3, 3, 512, 512])
No match: cylinder_3d_spconv_seg.upBlock1.trans_dilao.weight Shape in my_modeltorch.Size([256, 3, 3, 3, 512]).    Shape in checkpoint: torch.Size([3, 3, 3, 512, 256])
No match: cylinder_3d_spconv_seg.upBlock1.conv1.weight Shape in my_modeltorch.Size([256, 1, 3, 3, 256]).    Shape in checkpoint: torch.Size([1, 3, 3, 256, 256])
No match: cylinder_3d_spconv_seg.upBlock1.conv2.weight Shape in my_modeltorch.Size([256, 1, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 1, 3, 256, 256])
No match: cylinder_3d_spconv_seg.upBlock1.conv3.weight Shape in my_modeltorch.Size([256, 1, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 3, 3, 256, 256])
No match: cylinder_3d_spconv_seg.upBlock1.up_subm.weight Shape in my_modeltorch.Size([256, 3, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 3, 3, 256, 256])
No match: cylinder_3d_spconv_seg.upBlock2.trans_dilao.weight Shape in my_modeltorch.Size([128, 3, 3, 3, 256]).    Shape in checkpoint: torch.Size([3, 3, 3, 256, 128])
No match: cylinder_3d_spconv_seg.upBlock2.conv1.weight Shape in my_modeltorch.Size([128, 1, 3, 3, 128]).    Shape in checkpoint: torch.Size([1, 3, 3, 128, 128])
No match: cylinder_3d_spconv_seg.upBlock2.conv2.weight Shape in my_modeltorch.Size([128, 1, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 1, 3, 128, 128])
No match: cylinder_3d_spconv_seg.upBlock2.conv3.weight Shape in my_modeltorch.Size([128, 1, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 128])
No match: cylinder_3d_spconv_seg.upBlock2.up_subm.weight Shape in my_modeltorch.Size([128, 3, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 128])
No match: cylinder_3d_spconv_seg.upBlock3.trans_dilao.weight Shape in my_modeltorch.Size([64, 3, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 64])
No match: cylinder_3d_spconv_seg.upBlock3.conv1.weight Shape in my_modeltorch.Size([64, 1, 3, 3, 64]).    Shape in checkpoint: torch.Size([1, 3, 3, 64, 64])
No match: cylinder_3d_spconv_seg.upBlock3.conv2.weight Shape in my_modeltorch.Size([64, 1, 3, 3, 64]).    Shape in checkpoint: torch.Size([3, 1, 3, 64, 64])
No match: cylinder_3d_spconv_seg.upBlock3.conv3.weight Shape in my_modeltorch.Size([64, 1, 3, 3, 64]).    Shape in checkpoint: torch.Size([3, 3, 3, 64, 64])
No match: cylinder_3d_spconv_seg.upBlock3.up_subm.weight Shape in my_modeltorch.Size([64, 3, 3, 3, 64]).    Shape in checkpoint: torch.Size([3, 3, 3, 64, 64])
No match: cylinder_3d_spconv_seg.ReconNet.conv1.weight Shape in my_modeltorch.Size([64, 3, 1, 1, 64]).    Shape in checkpoint: torch.Size([3, 1, 1, 64, 64])
No match: cylinder_3d_spconv_seg.ReconNet.conv1_2.weight Shape in my_modeltorch.Size([64, 1, 3, 1, 64]).    Shape in checkpoint: torch.Size([1, 3, 1, 64, 64])
No match: cylinder_3d_spconv_seg.ReconNet.conv1_3.weight Shape in my_modeltorch.Size([64, 1, 1, 3, 64]).    Shape in checkpoint: torch.Size([1, 1, 3, 64, 64])
No match: cylinder_3d_spconv_seg.logits.weight Shape in my_modeltorch.Size([20, 3, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 20])
No match: cylinder_3d_spconv_seg.logits2.weight Shape in my_modeltorch.Size([3, 3, 3, 3, 128]).    Shape in checkpoint: torch.Size([3, 3, 3, 128, 3])
matched parameter sets: 227, and no matched: 49
